### PR TITLE
ci: add codeql as suggested by the bot (and fixes)

### PR DESCRIPTION
**Pull request by adrianreber**
This version of the PR from the bot also includes the corresponding fixes.

Same as #102 but with the corresponding fixes.

### CODE REVIEW

**Comment by adrianreber** [comment]
@rst0git PTAL for codeql integration

File: crit/Makefile [hunk 1]
```diff
@@ -10,12 +10,13 @@ proto_files := $(sort $(subst $(proto_path)/,,$(wildcard $(proto_path)/*.proto))
 # and replace all spaces with commas to use with go_opt flag
 comma := ,
 proto_opts := $(subst $() $(),$(comma),$(patsubst %,M%=$(import_path),$(proto_files)))
+branch := master
```
Comment (rst0git) [inline crit/Makefile:13]: Should we use `GIT_BRANCH` (upper case) for this constant instead?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -38,7 +38,13 @@ jobs:
       run: make -C scripts test
 
     - name: Test crit
-      run: sudo make -C test crit-test
+      run: |
+        make -C crit update-proto branch=${{ matrix.criu_branch }}
+        # If the definition of the protobuf files has changed upstream
+        # we just skip this test. Changed protobuf files will mean
+        # that crit-test will fail.
+        if [[ $(git status --porcelain | wc -l) -gt 0 ]]; then echo "Skipping crit-test because criu upstream changed protobuf definitions"; exit 0; fi
```
Comment (rst0git) [inline .github/workflows/main.yml:46]: Hmm, I'm not sure if this is a good idea.

We could just run `make gen-proto` and `crit-test` would pass even when the protobuf files have changed in the upstream repository:
```bash
make -C crit update-proto branch=${{ matrix.criu_branch }}
make -C crit gen-proto
sudo make -C test crit-test
```

Comment (adrianreber) [reply .github/workflows/main.yml:46]: Good idea

Comment (adrianreber) [reply .github/workflows/main.yml]: It was a good idea, but the protobuf version in Ubuntu 20.04 and 22.04 is too old to work with the files we generated with newer protobuf versions. Need to figure out how to best solve this.

**Comment by adrianreber** [comment]
Looks like a found a way to make it all work. @rst0git PTAL

**Review by rst0git — APPROVED** [review]
LGTM

**Pull request merged**